lib/generators/manageiq/plugin/plugin_generator.rb

```diff
@@ -54,6 +56,25 @@ def create_plugin_files
       empty_directory "spec/factories"
       empty_directory "spec/support"
       template "spec/spec_helper.rb"
+
+      if options[:js]
+        gitignore = <<~GITIGNORE
+
+          /node_modules/
+          /.pnp.*
+          /.yarn/*
+          !/.yarn/patches
+          !/.yarn/plugins
+          !/.yarn/releases
+          !/.yarn/sdks
+          !/.yarn/versions
+        GITIGNORE
+
+        inject_into_file ".gitignore", gitignore, :after => "/spec/manageiq\n"
+        template ".yarnrc.yml"
+        template "package.json"
+        # TODO template "yarn.lock"
```

Comment:
As discussed, let's template this since it's a basic shell with no actual packages in it.